hibernate-core/src/main/java/org/hibernate/engine/jdbc/BlobProxy.java

```diff
@@ -112,6 +112,9 @@ public byte[] getBytes(final long start, final int length) throws SQLException {
 		if ( length < 0 ) {
 			throw new SQLException( "Length must be great-than-or-equal to zero." );
 		}
+		if ( length == length()) {
+			return DataHelper.extractBytes( getBinaryStream() );
+		}
```

Comment:
This is maybe not a terrible idea, but it's also not quite right because `start` should also be checked.